src/app/reporting/Engine.cpp

```diff
@@ -492,7 +492,8 @@ CHIP_ERROR Engine::BuildAndSendSingleReportData(ReadHandler * apReadHandler)
     CHIP_ERROR err = CHIP_NO_ERROR;
     chip::System::PacketBufferTLVWriter reportDataWriter;
     ReportDataMessage::Builder reportDataBuilder;
-    chip::System::PacketBufferHandle bufHandle = System::PacketBufferHandle::New(chip::app::kMaxSecureSduLengthBytes);
+    size_t maxSduSize                          = chip::app::kMaxSecureSduLengthBytes;
+    chip::System::PacketBufferHandle bufHandle = nullptr;
```

Comment:
please declare this when initialized - having all variables at the top of the function is too much C-style API and hinders readability (we want the narrowest scope for variable as possible)

Reply:
There is a VerifyOrExit(..) on the readHandler and so I cannot declare this after that where it is being assigned from a ReadHandler function.